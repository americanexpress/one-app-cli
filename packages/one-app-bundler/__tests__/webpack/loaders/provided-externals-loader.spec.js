/*
 * Copyright 2019 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

const providedExternalsLoader = require('../../../webpack/loaders/provided-externals-loader');

jest.mock('loader-utils', () => ({
  getOptions: jest.fn(() => ({ providedExternals: ['ajv', 'lodash'] })),
}));

describe('provided-externals-loader', () => {
  it('should append the providedExternals to the default export', () => {
    const content = `\
import MyComponent from './components/MyComponent';
export default MyComponent;
`;
    expect(providedExternalsLoader(content)).toMatchSnapshot();
  });

  it('should throw an error when the wrong syntax is used - export from', () => {
    const content = `\
export default from './components/MyComponent';
`;
    expect(() => providedExternalsLoader(content)).toThrowErrorMatchingSnapshot();
  });

  it('should throw an error when the wrong syntax is used - module.exports', () => {
    const content = `\
module.exports = require('./components/MyComponent');
`;
    expect(() => providedExternalsLoader(content)).toThrowErrorMatchingSnapshot();
  });

  it('should throw an error when the wrong syntax is used - export default hoc()', () => {
    const content = `\
import SomeComponent from './SomeComponent';

export default hocChain(SomeComponent);
`;
    expect(() => providedExternalsLoader(content)).toThrowErrorMatchingSnapshot();
  });
});
