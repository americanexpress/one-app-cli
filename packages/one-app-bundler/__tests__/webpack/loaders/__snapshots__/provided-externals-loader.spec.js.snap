// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provided-externals-loader should append the providedExternals to the default export 1`] = `
"import MyComponent from './components/MyComponent';
export default MyComponent;
;
MyComponent.appConfig = Object.assign({}, MyComponent.appConfig, {
  providedExternals: {
    'ajv': { version: '6.12.6', module: require('ajv')},
  'lodash': { version: '4.17.20', module: require('lodash')},
  },
});

if(global.getTenantRootModule === undefined || (global.rootModuleName && global.rootModuleName === 'undefined')){
global.getTenantRootModule = () => MyComponent;
global.rootModuleName = 'undefined';
}
"
`;

exports[`provided-externals-loader should throw an error when the wrong syntax is used - export default hoc() 1`] = `"@americanexpress/one-app-bundler: Module must use \`export default VariableName\` in index syntax to use providedExternals"`;

exports[`provided-externals-loader should throw an error when the wrong syntax is used - export from 1`] = `"@americanexpress/one-app-bundler: Module must use \`export default VariableName\` in index syntax to use providedExternals"`;

exports[`provided-externals-loader should throw an error when the wrong syntax is used - module.exports 1`] = `"@americanexpress/one-app-bundler: Module must use \`export default VariableName\` in index syntax to use providedExternals"`;
